processing/src/main/java/io/druid/query/BaseQuery.java

```diff
@@ -105,26 +106,31 @@ public static void checkInterrupted()
   }
 
   public static final String QUERYID = "queryId";
+
   private final DataSource dataSource;
   private final boolean descending;
   private final Map<String, Object> context;
   private final QuerySegmentSpec querySegmentSpec;
+  @Nullable
+  private final QueryMetrics<?> queryMetrics;
   private volatile Duration duration;
 
   public BaseQuery(
       DataSource dataSource,
       QuerySegmentSpec querySegmentSpec,
       boolean descending,
-      Map<String, Object> context
+      Map<String, Object> context,
+      QueryMetrics<?> queryMetrics
```

Comment:
can we keep a constructor with the same signature for the sake of query extensions?